main.py

```diff
@@ -31,11 +38,194 @@ def setup(self):
 
         So, if you have any setup routine, insert it here.
         """
+        self.basic_pipeline = BASIC_PIPELINE
+        self.subscribed_napps = SUBSCRIBED_NAPPS
         self.pipeline_controller = self.get_pipeline_controller()
+        self._pipeline_lock = Lock()
+        self.required_napps = set()
+        self.load_enable_table(self.get_enabled_table())
 
     def execute(self):
         """Execute once when the napp is running."""
 
+    def get_enabled_table(self):
```

Comment:
If you could also add the returned typing annotation, that'd be great, cosmetic detail since we don't have a linter for that yet, but it does enhance code readability out of the gate.

Reply:
Added annotations.